### ISSUE TITLE
FPS Fall in Mortal Kombat Gold

### ISSUE THREAD

**Issue by spielvan** [reicast/reicast-emulator#1469]
**Please Note: This form is the minimum required information for submitting bugs.**  
**Removing this form may lead to your issue being closed until it is completed.**

| Platform | Branch | Hash | CIDL |
| ------ | ------ | ---- | ---- |
| Installed OS and version (ie. Android 8.0) | Branch (or Google Play) | Hashtag (skip if unknown) | Downloaded from build site? |
MX3-G android tv box 32bits, Mali 450 Open Gl 2.0, with android 5.1.1 Lollipop
Build fh/mymaster Commit
10d9761 2018-12-12T13: 11: 12,000Z

**_Description of the Issue_**

The game Mortal Kombat Gold works perfectly, but in the part of the "Videos CGs" fps drops to 22 fps leaving the videos reading, during the gameplay the game is perfect with 60 fps and the other problem is the memory card of him that is not enabled , then save in the game but when you exit the game and enter again the save is not saved, making it impossible to review the videos saved in theater mode.

Add a short, concise description of the issue here

**_Debugging Steps Tested_**

  * Fill in any steps already tried here
  * Begin each new line with an asterisk
  * If no steps are required, please skip

**_Logs Gathered_**

```

Please paste the contents of the log / logs here
You may leave this field blank if you have none.

```

**_Screenshots_**

(Replace this line with any issue screenshots)

- [ ] Place an X in the box to confirm you have filled in this form

**Comment by dmiller423** [comment]
@flyinghead the report is about your branch,  I'm fairly sure the problem likely affects both but I haven't tested.  I'm assuming the save issue is file related,  video lag could be aica or several things.

**Comment by flyinghead** [comment]
FMV videos are played at 30 fps max, even though the game runs at 60 fps.
On my test phone, MKG videos do play at 30 fps but it is a recent device (Snapdragon 845).

It is true that reicast is rather slow at decoding mpeg video. I'm not sure why.

**Comment by spielvan** [comment]
What made me confused is that this same android box, with the build master bbc54e4 2018-12-01T19: 02: 16.000Z, the Mortal Kombat Gold videos run at 30 or 31 fps.

**Comment by flyinghead** [comment]
One significant difference between my branch and master is that YUV textures (used by FMV videos) are converted to 8888 format while master downsamples to 4444. So textures are twice as big and this probably has an impact on lower-end hardware.

**Comment by MrPsyMan** [comment]
Yes. Older hardware has a hard time with things like data throughput, scaling, modifier volumes, render to texture and clipping (amongst other things). 

Most of the things unimplemented from nulldc and commented out on the source are like that because of the observed side-effects leading to slowdowns.

**Comment by flyinghead** [comment]
Being able to disable modifier volumes and clipping in the user prefs could help in some situations.

However it won't help in this case as this is FMV video playing.
